pkg/kubeadm/images/images.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package images
+
+import (
+	"fmt"
+	"runtime"
+)
+
+const (
+	KubeEtcdImage = "etcd"
+
+	KubeApiServerImage         = "apiserver"
+	KubeControllerManagerImage = "controller-manager"
+	KubeSchedulerImage         = "scheduler"
+	KubeProxyImage             = "proxy"
+
+	KubeDnsImage         = "kube-dns"
+	KubeDnsmasqImage     = "dnsmasq"
+	KubeExechealthzImage = "exechealthz"
+
+	gcrPrefix   = "gcr.io/google_containers"
+	etcdVersion = "2.2.5"
+	kubeVersion = "v1.4.0-alpha.3"
```

Comment:
> but would there be a better way to gather these constants so we don't have to make sure that it's changed for each k8s release?

Eventually, it should be an automated change on each release.

> With etcd, dns, etc. is there a way to change what versions of said software is used? (config file, env variables, etc?)

etcd and DNS are vital parts in a Kubernetes cluster, so I'm in favor of this tool to be highly opinionated on what images and versions of images are used, and do not allow for user manipulation.
For other add-ons, I know it will be different, and you'll be able to provide them as manifests (`kubectl apply -f`).